sources/Interop/Hid/shared/hidpi/HIDP_PREPARSED_DATA.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+// Ported from shared/hidpi.h in the Windows SDK for Windows 10.0.18362.0
+// Original source is Copyright © Microsoft. All rights reserved.
+
+namespace TerraFX.Interop
+{
+    public partial struct HIDP_PREPARSED_DATA
```

Comment:
Looks like this is an opaque struct and is meant to be essentially a `handle` (`IntPtr`), I can only find a typedef such as `typedef struct _HIDP_PREPARSED_DATA * PHIDP_PREPARSED_DATA;` and no formal definition of the actual struct.

Reply:
From what I read in a Code Project sample I've been using, the data is internal and only intended to be used by the system.
> The Raw Input API gives us access to some device information using the GetRawInputDeviceInfo function. Given the handle to the raw input device and the RIDI_PREPARSEDDATA command we can obtain so called "preparsed data". The WDK documentation just says, "The internal structure of a _HIDP_PREPARSED_DATA structure is reserved for internal system use".
https://www.codeproject.com/articles/185522/using-the-raw-input-api-to-process-joystick-input

I know this might not be authoritative but it's all I have to go on. Let me know how you'd like to handle this one.

Reply:
I would favor keeping it since it's a type logically, even if not physically. From my exploration it seems the way one is supposed to allocate space for this type is to allocate a byte array and then cast the type. It's that oddball.

Reply:
I guess by "opaque struct" you're maybe identifying a concept I'm not that familiar with. Maybe this pattern isn't so oddball. Let me know if you'd like me to map it to `IntPtr` and I will.

Reply:
Right, it's meant to be completely "opaque". There are similar in other libraries (`XDisplay` on `X11`) and the general point is to the average user it is just a handle, but to a sufficiently advanced user you could try and access the underlying bits (you can do the same with `HWND` which, depending on your defines is either a typedef for a `void*` or a typedef to an opaque `struct __HWND`).

The general convention for these is to make it an `IntPtr` and the dev can define and cast the pointer to their own type if they really want to access internals.

Reply:
By "opaque" I mean it is a struct that completely hides the internals from the user (in this case, it exposes nothing in the header and you only have a pointer to it, which you can pass to other methods) vs a "transparent" one where you can see, access, and depend on the members directly.

Reply:
https://en.wikipedia.org/wiki/Opaque_pointer was a good read on the topic. Thanks for pointing this out.